herbert_core/classes/instrument_classes/instrument/detectors/@IX_detector_array/IX_detector_array.m

```diff
@@ -74,205 +128,412 @@
                     end
+                    
+                else
+                    % Delegate processing of varargin to IX_detector_bank.
+                    % This implies that varargin is the whole set of
+                    % detector bank constructor arguments.
+                    obj.det_bank_ = IX_detector_bank (varargin{:});
                 end
             end
-
+            
         end
+        
         %------------------------------------------------------------------
-        % Get methods for dependent properties
-
-        function val = get.filename(obj)
-            val = obj.filename_;
+        % Set methods for dependent properties
+        %------------------------------------------------------------------
+        
+        % Mirrors of private properties; these define object state:
+        % ---------------------------------------------------------
+        function obj = set.det_bank (obj, val)
+            if ~isa(val,'IX_detector_bank') || isempty(val)
+                error('HERBERT:IX_detector_array:invalid_argument',...
+                    'Detector bank(s) must be a non-empty IX_detector_bank array')
+            end
+            obj.det_bank_ = val(:);
+            if obj.do_check_combo_arg_
+                obj = obj.check_combo_arg();
+            end
         end
-
-        function obj = set.filename(obj,val)
+        
+        %---------------------------
+        function obj = set.filepath (obj, val)
+            if isempty(val)
+                val = '';
+            elseif ~is_string(val)
+                error('HERBERT:IX_detector_array:invalid_argument',...
+                    'Filepath must be a string ')
+            end
+            obj.filepath_ = val;
+            if obj.do_check_combo_arg_
+                obj = obj.check_combo_arg();
+            end
+        end
+        
+        %---------------------------
+        function obj = set.filename (obj, val)
+            if isempty(val)
+                val = '';
+            elseif ~is_string(val)
+                error('HERBERT:IX_detector_array:invalid_argument',...
+                    'Filename must be a string ')
+            end
             obj.filename_ = val;
+            if obj.do_check_combo_arg_
+                obj = obj.check_combo_arg();
+            end
         end
-
-        function val = get.filepath(obj)
-            val = obj.filepath_;
+        
+        % Generic properties across all detector banks:
+        % ---------------------------------------------
+        function obj = set.id (obj, val)
+            ndets = arrayfun (@(O)(O.ndet), obj.det_bank_);
+            % Input must be an array with ndet elements
+            if ~isnumeric(val) || numel(val)~=sum(ndets)
+                error('HERBERT:IX_detector_array:invalid_argument',...
+                    ['Detector indices must be a numeric array with ',...
+                    num2str(sum(ndets)), ' elements'])
```

Comment:
```suggestion
                    'Detector indices must be a numeric array with %d elements', ...
                    sum(ndets))
```

Reply:
Done. It makes things more readable; also changed in other similar places.